src/main/java/com/ea/restaurant/service/impl/Oauth2ServiceImpl.java

```diff
@@ -17,10 +18,15 @@
 import com.ea.restaurant.service.Oauth2Service;
 import com.ea.restaurant.utils.Oauth2Util;
 import com.nimbusds.jose.JOSEException;
+import com.nimbusds.jose.proc.BadJOSEException;
+import java.text.ParseException;
+import java.util.Date;
 import org.springframework.beans.factory.annotation.Value;
 import org.springframework.security.crypto.bcrypt.BCrypt;
 import org.springframework.stereotype.Service;
+import org.springframework.transaction.annotation.Transactional;
 
+@Transactional
```

Comment:
`@Transactional` is more beneficial to use it over functions instead so that you can setup your transaction depending on the operations it will do. Update the existing methods as well.

Reply:
Done.

Reply:
Done.